### ISSUE TITLE
problem with node v0.12.2

### ISSUE THREAD

**Issue by ApolloTang** [node-inspector/node-inspector#603]
update to node v0.12.2 resulted in:

Error: Cannot find module '/usr/local/lib/node_modules/node-inspector/node_modules/v8-debug/build/debug/v0.3.4/node-v14-darwin-x64/debug.node'

$ ls /usr/local/lib/node_modules/node-inspector/node_modules/v8-debug/build/debug
v0.4.2

$ npm ls | grep node-inspector
v8-debug@0.4.2 /usr/local/lib/node_modules/node-inspector/node_modules/v8-debug

$ node -v
v0.12.2

:(

$ node-inspector -v
Node Inspector v0.9.2

**Comment by kureus** [comment]
+1 @ApolloTang Found a workaround for this? Google doesn't instantly help

**Comment by ApolloTang** [comment]
$ nodemon debug myfile.js 
is my work around for now :( 
let me know if you figure this out..

**Comment by fetis** [comment]
You need to update to version 0.10

**Comment by akirattii** [comment]
node-inspector v0.10.0 on nodejs v0.12.2 stable does not work...

```
$ node -v
v0.12.2
$ node-inspector -v
Node Inspector v0.10.0

$ node-debug app.js 
Node Inspector is now available from http://127.0.0.1:8080/debug?ws=127.0.0.1:8080&port=5858
Debugging `app.js`

Debugger listening on port 5858
```

Then the Chrome browser's window opens automatically and shows the following error on its console  tab.

```
Error: Cannot find module '/usr/local/lib/node_modules/node-inspector/node_modules/v8-debug/build/debug/v0.4.2/node-v14-linux-x64/debug.node'
Runtime.getProperties failed.
Object #-3# not found
```

**Comment by 3y3** [comment]
@akirattii , this looking like a problem with v8-debug submodule. I saw this problem then change my node version from 0.12.0 to 0.12.2. Can you try to completely reinstall node-inspector:
```bash
npm uninstall -g node-inspector
npm install -g node-inspactor
```

**Comment by akirattii** [comment]
Hey @3y3 , thank you for telling.
I've tried as you write, but the installation is failed. 

$ sudo npm uninstall -g node-inspector
unbuild node-inspector@0.10.0
$ sudo npm install -g node-inspector
&gt; ws@0.4.32 install /usr/local/lib/node_modules/node-inspector/node_modules/ws
&gt; (node-gyp rebuild 2> builderror.log) || (exit 0)

Is there any workarounds?

**Comment by akirattii** [comment]
@3y3 , I retried the reinstall again and waited for a long time, then it's been finished with some warnings.

```
$ sudo npm install -g node-inspector

&gt; ws@0.4.32 install /usr/local/lib/node_modules/node-inspector/node_modules/ws
&gt; (node-gyp rebuild 2&gt; builderror.log) || (exit 0)

make: ディレクトリ `/usr/local/lib/node_modules/node-inspector/node_modules/ws/build' に入ります
  CXX(target) Release/obj.target/bufferutil/src/bufferutil.o
  SOLINK_MODULE(target) Release/obj.target/bufferutil.node
  SOLINK_MODULE(target) Release/obj.target/bufferutil.node: Finished
  COPY Release/bufferutil.node
  CXX(target) Release/obj.target/validation/src/validation.o
  SOLINK_MODULE(target) Release/obj.target/validation.node
  SOLINK_MODULE(target) Release/obj.target/validation.node: Finished
  COPY Release/validation.node
make: ディレクトリ `/usr/local/lib/node_modules/node-inspector/node_modules/ws/build' から出ます
npm WARN optional dep failed, continuing default-browser-id@1.0.2
npm WARN engine xmlbuilder@2.2.1: wanted: {"node":"0.8.x || 0.10.x"} (current: {"node":"0.12.2","npm":"2.7.4"})

&gt; v8-profiler@5.2.4 install /usr/local/lib/node_modules/node-inspector/node_modules/v8-profiler
&gt; node-pre-gyp install --fallback-to-build

[v8-profiler] Success: "/usr/local/lib/node_modules/node-inspector/node_modules/v8-profiler/build/profiler/v5.2.4/node-v14-linux-x64/profiler.node" is installed via remote

&gt; v8-debug@0.4.2 install /usr/local/lib/node_modules/node-inspector/node_modules/v8-debug
&gt; node-pre-gyp install --fallback-to-build

[v8-debug] Success: "/usr/local/lib/node_modules/node-inspector/node_modules/v8-debug/build/debug/v0.4.2/node-v14-linux-x64/debug.node" is installed via remote
/usr/local/bin/node-debug -&gt; /usr/local/lib/node_modules/node-inspector/bin/node-debug.js
/usr/local/bin/node-inspector -&gt; /usr/local/lib/node_modules/node-inspector/bin/inspector.js
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/npmconf requires mkdirp@'~0.3.3' but will load
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/mkdirp,
npm WARN unmet dependency which is version 0.5.0
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/npmconf requires osenv@'0.0.3' but will load
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/osenv,
npm WARN unmet dependency which is version 0.1.0
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/npmconf requires nopt@'2' but will load
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/nopt,
npm WARN unmet dependency which is version 3.0.1
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/npmconf requires semver@'2' but will load
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/semver,
npm WARN unmet dependency which is version 4.3.1
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/npmconf requires ini@'~1.1.0' but will load
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/ini,
npm WARN unmet dependency which is version 1.3.3
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/read-package-json/node_modules/normalize-package-data requires github-url-from-git@'~1.1.1' but will load
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/github-url-from-git,
npm WARN unmet dependency which is version 1.4.0
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/read-package-json/node_modules/normalize-package-data requires github-url-from-username-repo@'^0.1.0' but will load
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/github-url-from-username-repo,
npm WARN unmet dependency which is version 1.0.2
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/read-package-json/node_modules/normalize-package-data requires semver@'2' but will load
npm WARN unmet dependency /usr/local/lib/node_modules/npm/node_modules/semver,
npm WARN unmet dependency which is version 4.3.1
node-inspector@0.10.0 /usr/local/lib/node_modules/node-inspector
├── which@1.0.9
├── async@0.9.0
├── semver@3.0.1
├── debug@1.0.4 (ms@0.6.2)
├── yargs@1.3.3
├── strong-data-uri@0.1.1 (truncate@1.0.4)
├── rc@0.5.5 (strip-json-comments@0.1.3, deep-extend@0.2.11, ini@1.3.3, minimist@0.0.10)
├── serve-favicon@2.2.0 (fresh@0.2.4, ms@0.7.0, parseurl@1.3.0, etag@1.5.1)
├── glob@4.5.3 (inherits@2.0.1, inflight@1.0.4, once@1.3.1, minimatch@2.0.4)
├── express@4.12.3 (merge-descriptors@1.0.0, utils-merge@1.0.0, cookie-signature@1.0.6, methods@1.1.1, cookie@0.1.2, fresh@0.2.4, escape-html@1.0.1, range-parser@1.0.2, finalhandler@0.3.4, content-type@1.0.1, vary@1.0.0, parseurl@1.3.0, serve-static@1.9.2, content-disposition@0.5.0, path-to-regexp@0.1.3, depd@1.0.1, qs@2.4.1, debug@2.1.3, etag@1.5.1, on-finished@2.2.1, send@0.12.2, proxy-addr@1.0.7, accepts@1.2.5, type-is@1.6.1)
├── biased-opener@0.2.5 (opener@1.4.1, minimist@1.1.0, x-default-browser@0.3.0, browser-launcher2@0.4.5)
├── ws@0.4.32 (tinycolor@0.0.1, options@0.0.6, commander@2.1.0, nan@1.0.0)
├── v8-profiler@5.2.4 (nan@1.5.3)
└── v8-debug@0.4.2 (nan@1.8.4)
```

Now I've tried to node-debug...

```
$ node-debug app.js 
Node Inspector is now available from http://127.0.0.1:8080/debug?ws=127.0.0.1:8080&port=5858
Debugging `app.js`

Debugger listening on port 5858
```

But I could not debug the app.js because the webserver seems not to stand-by (localhost:3000 was down.) . 
The browser's console shows a following error.

```
Runtime.getProperties failed.
Object #-3# not found
```

**Comment by kureus** [comment]
@3y3 Running node 0.12.2 and node-inspector 0.10.0 Tried a full removal and reinstall of node inspector as suggested. On node-debug app.js still receive 

<code> Error: Cannot find module '/usr/local/lib/node_modules/node-inspector/node_modules/v8-debug/build/debug/v0.4.2/node-v14-linux-x64/debug.node' </code>

Anything else I can do to provide more context?

**Comment by 3y3** [comment]
@kureus , I wonder to look on your install log.

**Comment by kureus** [comment]
@3y3 Ok, digging into this a little I actually have this in /usr/local

> /usr/local/lib/node_modules/node-inspector/node_modules/v8-debug/build/debug/v0.4.2/node-v11-linux-x64/debug.node

whereas node-inspector is expecting to find 

> /usr/local/lib/node_modules/node-inspector/node_modules/v8-debug/build/debug/v0.4.2/node-v14-linux-x64/debug.node

v11 vs v14?

**Comment by 3y3** [comment]
@kureus , thank. Your problem confirmed as bug. I need to update `v8-debug` and `v8-progiler` bundled `node-pre-gyp` version to 0.6.5. This version knows about node 0.12.2.

**Comment by 3y3** [comment]
Please try to completely reinstall node-inspector now. I updated `v8-debug` and `v8-profiler` versions to be compatible with node 0.12.2 and latest iojs

**Comment by ashelley** [comment]
Hello, I am experiencing this problem too:

- Run my program with node-debug -c server.js 
or  node --debug-brk server.js, then in another tab run node-inspector

Node inspector hangs with one error:

```
Unknown experiment canvasInspection
```

I've waited up to 5 minutes after starting to see if "anything happens" but the chrome tab is locked up and you have to kill the chrome tab, 'waiting' doesn't seem to help.  

I am posting here as this issue https://github.com/node-inspector/node-inspector/issues/534 says to reopen if you are experiencing issues.

I've attempted multiple uninstall / reinstalls of node-inspector on windows 8.1 and kubuntu 14.04 using nvm to install node inspector 0.10 and node stable 0.12.2.

In summary, other than chrome tab locking up and the one console.log error in the chrome tab I have no other indications of failure and the node program I am debugging does not continue so that it ends up listening on a port but does not crash with an error.

If anyone has any insight in fixing this issue please let me know.  Also If anyone wants me to perform some procedures to attempt to work around or fix the issue let me know.

-Adam

**Comment by 3y3** [comment]
@ashelley , thanks for verbose description. At first glance this is a problem with `ws` module. I wonder to see your install log. 
You also can try to update `ws` module
```
cd node_inspector_folder
npm install ws@0.7.1
```

Next information that can be useful is [debug logs](https://github.com/node-inspector/node-inspector/wiki/Debug-logs)

**Comment by ashelley** [comment]
So after looking at this a bit more I don't necessarily think that the problem is related to the ws version.  I actually got node-inspector to load on the front end but I believe there is a problem with how node-inspector decides which files to load which is the cause of problem.  However, the following is a result of testing with ws@0.7.1 installed:

I ended up waiting about 10 minutes this time and this is what I observed.  Node inspector starts eating memory and hits my memory limit and starts thrashing the GPU (probably related to gc'ing do to memory usage).
```
top - 09:06:14 up 23:19,  5 users,  load average: 1.13, 1.79, 1.27
Tasks:   2 total,   1 running,   1 sleeping,   0 stopped,   0 zombie
%Cpu(s): 96.3 us,  3.0 sy,  0.0 ni,  0.0 id,  0.0 wa,  0.0 hi,  0.7 si,  0.0 st
KiB Mem:   2050036 total,  1934440 used,   115596 free,    54776 buffers
KiB Swap:  2097148 total,   409272 used,  1687876 free.    86888 cached Mem

  PID USER      PR  NI    VIRT    RES    SHR S %CPU %MEM     TIME+ COMMAND                                                       
12709 adam      20   0 1816356 959456   5044 R 93.5 46.8   4:53.47 node   <--- node-inspector                                                       
12572 adam      20   0 1251264  20532   1340 S  0.0  1.0   0:00.39 node             
```

After looking at the debug logs I think i know what the problem is:

There is a line in it that starts like this in the debug log:
```
  node-inspector:protocol:devtools backend: {"id":8,"result":{"frameTree"
...
<content skipped (picture the output of directory listing of lots of javascript files in json format>
...
"type":"Script","mimeType":"text/javascript"}]}}} +6m
```

This is where node-inspector goes crazy... I ran node-inspector from a directory like this:

```
:/var/www/test/my-test-project$ DEBUG=node-inspector* node-inspector --web-host 192.168.1.50
```

For some reason node-inspector tried to load every javascript file from the directory above where it's working, see this screenshot:
```
![image](https://cloud.githubusercontent.com/assets/3195045/7477321/e70a7486-f307-11e4-8577-8db5f618e8b6.png)
```
Note that it's loading files from /var/www/test instead of /var/www/test/my-test/project.  I think I believe the fact that node-inspector is loading files from the wrong directory (one level up) which is causing the problem.  In the above directory i have 1000's and 1000's of javascript files that node inspector sends the directory listing to the front end.  

However, note that this isn't the slow part... just above the problematic directory loading there are these entries:
```
  node-inspector:protocol:devtools frontend: {"id":20,"method":"IndexedDB.enable"} +42s
  node-inspector:protocol:devtools backend: {"id":20} +0ms
  node-inspector:protocol:devtools frontend: {"id":21,"method":"HeapProfiler.enable"} +1m
  node-inspector:protocol:devtools backend: {"id":21} +56ms
```

Perhaps this is related to the above listed problem (i'm not sure).

In summary, I think there's a bug somewhere in node-inspector in deciding what source files to send to the frontend.

**Comment by ashelley** [comment]
Update:

Maybe this is my fault?.. i need to diagnose this more... I had no package.json in my current directory from where I ran my script:

```
:/var/www/test/my-test-project$ node --debug-brk server.js
```

After adding a package.json file to this directory node-inspector behaved correctly. Oooops.  I also reverted the version of ws to the included version of 0.10 and had no issue.  

@3y3 sorry for wasting your time.  I didn't realize that package.json needed to exist in the directory in which you run your node script from.

-Adam

**Comment by 3y3** [comment]
@ashelley , no problem. You can also use `--no-preload` flag, if you want to work in directory without `package.json`

**Comment by 3y3** [comment]
I close issue because work on 0.12.2 confirmed by @ashelley .
If original problem wasn't resolved, call me to reopen this issue.

**Comment by bobpappas** [comment]
This issue should be reopened. I am getting the same problem:
Error: Cannot find module '/usr/local/lib/node_modules/node-inspector/node_modules/v8-debug/build/debug/v0.4.3/node-v43-darwin-x64/debug.node'

$ node-inspector -v
Node Inspector v0.10.0
$ node -v
v0.12.2

in the directory above, only node-v14-darwin-x64 is listed. Not v43.

**Comment by bobpappas** [comment]
When renaming the directory to node-v43-darwin-x64, I get:
Error: Module version mismatch. Expected 43, got 14.

**Comment by 3y3** [comment]
@bobpappas , you have `iojs` installed anywhere?

**Comment by bobpappas** [comment]
Yeah, I'm trying to use electron and their instructions for using node-inspector. Why is io.js causing issues?

**Comment by martellalex** [comment]
I was running into this issue using iTerm 2 on Mac OSX.  I had one terminal window running `$node-inspector` and another where I was doing `$nodemon --debug-brk main.js`.  The problem turned out to be that on one terminal window I was using an old version of node (which is why I was getting the error with node-inspector), and on another window I was using the latest stable version of node (which is why I thought I had everything set up correctly).

So solution was to make sure I was using the latest stable version (e.g. `$nvm use stable`) of node on BOTH terminal windows.
